### PR TITLE
[React] Upgrade webpack-dev-server 4.1.0

### PR DESCRIPTION
**Pull request by qmonmert**
Always an error

image

cc @deepu105 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/webpack/webpack.dev.js.ejs [hunk 1]
```diff
@@ -83,9 +84,6 @@ module.exports = async (options) => webpackMerge(await commonConfig({ env: ENV }
       target: 'ws://127.0.0.1:<%= serverPort %>',
       ws: true
     }<% } %>],
-    watchOptions: {
```
Comment (deepu105) [inline generators/client/templates/react/webpack/webpack.dev.js.ejs:86]: Are node modules automatically ignored now?

Comment (qmonmert) [reply generators/client/templates/react/webpack/webpack.dev.js.ejs:86]: it seems https://github.com/webpack/webpack-dev-server/blob/master/migration-v4.md

**Review by mraible — APPROVED** [review]

**Comment by mshima** [comment]
Forgot to link https://github.com/jhipster/generator-jhipster/pull/16150.
Tracked after bisect the react image problem.

**Comment by qmonmert** [comment]
so my PR is useless?

**Comment by mshima** [comment]
@qmonmert I have based the webpack changes on yours.
I was going to merge this PR before merging my, but last commit would conflict with -- trailing changes.
Thanks for your PR.
Closing.

**Comment by deepu105** [comment]
@qmonmert I have added a bounty to this so please claim the same and thank you for tracking this

**Comment by qmonmert** [comment]
thanks @deepu105 :)

@pascalgrimaud & @jdubois I claim my money https://opencollective.com/generator-jhipster/expenses/49389

**Pull request closed without merging**